Sources/SwiftParser/Nominals.swift

```diff
@@ -228,12 +229,16 @@ extension Parser {
             )
           )
         } else {
+          if self.currentToken.starts(with: "~") {
+            withoutToken = self.consumePrefix("~", as: .prefixOperator)
+          }
```

Comment:
Is the `~` part of the identifier? If so, that would surprise me. I would assume that you should be able to do 

```swift
withoutToken = self.consumeIfContextualPunctuator("~")
```